src/ripple/rpc/impl/TransactionSign.cpp

```diff
@@ -745,16 +752,64 @@ Json::Value checkMultiSignFields (Json::Value const& jvRequest)
     if (!tx_json.isMember (jss::Sequence))
         return RPC::missing_field_error ("tx_json.Sequence");
 
-    if (!tx_json.isMember ("SigningPubKey"))
+    if (!tx_json.isMember (sfSigningPubKey.getJsonName()))
         return RPC::missing_field_error ("tx_json.SigningPubKey");
 
-    if (!tx_json["SigningPubKey"].asString().empty())
+    if (!tx_json[sfSigningPubKey.getJsonName()].asString().empty())
         return RPC::make_error (rpcINVALID_PARAMS,
             "When multi-signing 'tx_json.SigningPubKey' must be empty.");
 
     return Json::Value ();
 }
 
+// Sort and validate an stSigners array.
+//
+// Returns a null Json::Value if there are no errors.
+static Json::Value sortAndValidateSigners (
+    STArray& signers, AccountID const& signingForID)
+{
+    if (signers.empty ())
+        return RPC::make_param_error ("Signers array may not be empty.");
+
+    // Signers must be sorted by Account.
+    std::sort (signers.begin(), signers.end(),
+        [](STObject const& a, STObject const& b)
+    {
+        return (a.getAccountID (sfAccount) < b.getAccountID (sfAccount));
```

Comment:
I'm kind of partial to:  `return a[sfAccount] < b[sfAccount];`

Just throwing it out there